src/stakeinput.cpp

```diff
@@ -121,21 +121,7 @@ CDataStream CPivStake::GetUniqueness() const
 //The block that the UTXO was added to the chain
 CBlockIndex* CPivStake::GetIndexFrom()
 {
-    if (pindexFrom)
-        return pindexFrom;
-    uint256 hashBlock = UINT256_ZERO;
-    CTransaction tx;
-    if (GetTransaction(opOutpointFrom->hash, tx, hashBlock, true)) {
-        // If the index is in the chain, then set it as the "index from"
-        if (mapBlockIndex.count(hashBlock)) {
-            CBlockIndex* pindex = mapBlockIndex.at(hashBlock);
-            if (chainActive.Contains(pindex))
-                pindexFrom = pindex;
-        }
-    } else {
-        LogPrintf("%s : failed to find tx %s\n", __func__, opOutpointFrom->hash.GetHex());
-    }
-
+    if (!pindexFrom) throw std::runtime_error("CPivStake: uninitialized pindexFrom");
```

Comment:
(1885e91ac65d8d17259ec4695117a7f0879bf8c8)
Guess that the rational is: we have to go through `InitFromTxIn` before calling this function?
That's good, but  maybe better to add a comment.
Also make the function `const`.
Also update the callers where
```cpp
CBlockIndex* pindexFrom = stake->GetIndexFrom();
if (!pindexFrom) {
    // some error
}
```